tensorflow-keras/src/test/java/org/tensorflow/keras/optimizers/AdamTest.java

```diff
@@ -0,0 +1,263 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+=======================================================================*/
+package org.tensorflow.keras.optimizers;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
+import org.tensorflow.Tensor;
+import static org.tensorflow.framework.optimizers.Adam.FIRST_MOMENT;
+import static org.tensorflow.framework.optimizers.Adam.SECOND_MOMENT;
+import org.tensorflow.framework.optimizers.Optimizer;
+import static org.tensorflow.keras.optimizers.Adam.BETA_ONE_DEFAULT;
+import static org.tensorflow.keras.optimizers.Adam.BETA_ONE_KEY;
+import static org.tensorflow.keras.optimizers.Adam.BETA_TWO_DEFAULT;
+import static org.tensorflow.keras.optimizers.Adam.BETA_TWO_KEY;
+import static org.tensorflow.keras.optimizers.Adam.EPSILON_DEFAULT;
+import static org.tensorflow.keras.optimizers.Adam.EPSILON_KEY;
+import static org.tensorflow.keras.optimizers.Adam.LEARNING_RATE_DEFAULT;
+import static org.tensorflow.keras.optimizers.Adam.LEARNING_RATE_KEY;
+import static org.tensorflow.keras.optimizers.OptimizerInterface.NAME_KEY;
+import org.tensorflow.keras.utils.ND;
+import org.tensorflow.keras.utils.TestSession;
+import org.tensorflow.ndarray.FloatNdArray;
+import org.tensorflow.ndarray.NdArrays;
+import org.tensorflow.ndarray.Shape;
+import org.tensorflow.op.Op;
+import org.tensorflow.op.Ops;
+import org.tensorflow.op.core.Assign;
+import org.tensorflow.op.core.Constant;
+import org.tensorflow.op.core.Variable;
+import org.tensorflow.types.TFloat32;
+
+/** Test cases for Adam Optimizer */
+public class AdamTest {
+  private TestSession.Mode tf_mode = TestSession.Mode.GRAPH;
+
+  int index;
+
+  public AdamTest() {}
+
+  @BeforeAll
+  public static void setUpClass() {}
+
+  @AfterAll
+  public static void tearDownClass() {}
+
+  @BeforeEach
+  public void setUp() {}
+
+  @AfterEach
+  public void tearDown() {}
+
+  /** Test of create method, of class Adam. */
+  @Test
+  public void testCreate() {
+    try (TestSession testSession = TestSession.createTestSession(tf_mode)) {
+      Ops tf = testSession.getTF();
+      Map<String, Object> config = new HashMap<>();
+      config.put(NAME_KEY, "AdaDelta");
+      config.put(LEARNING_RATE_KEY, LEARNING_RATE_DEFAULT);
+      config.put(BETA_ONE_KEY, BETA_ONE_DEFAULT);
+      config.put(BETA_TWO_KEY, BETA_TWO_DEFAULT);
+      config.put(EPSILON_KEY, EPSILON_DEFAULT);
+      AdaDelta expResult = new AdaDelta(tf);
+      AdaDelta result = AdaDelta.create(tf, config);
+      assertEquals(expResult.getConfig(), result.getConfig());
+    }
+  }
+
+  @Test
+  public void testBasic() {
```

Comment:
(I wouldn't propose holding up this PR for this issue.)

This test raises interesting questions about both our API layers and our testing strategy. It purports to be a test of `keras.optimizers.Adam`, but in fact it tests the actual underlying Adam logic that is exposed by `framework.optimizers.Adam` -- which is not, however, yet tested in `framework.optimizers`. 

If we stick with the current layering strategy, I'd propose that in a future PR we move this test into `framework.optimizers`.

Reply:
As of now, I would vote that these tests be moved to framework, especially in that we are moving an addition 3 optimizers to framework that are defined in Keras.  Also, the JavaDoc on the existing Optimizers need a lot of work.

Reply:
Given that I have limited time availability and am a decent technical writer, that's a likely area for me to contribute.